### ISSUE TITLE
L034 issue if a jinja variable is defined

### ISSUE THREAD

**Issue by olivierdet** [sqlfluff/sqlfluff#4155]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following code doesn't raise the 034 issue. 
```
    select
        case_event_id
        , case_id
        , created_at as owner_from
        , coalesce(lead(created_at, 1) over (
            partition by case_id order by event_order), '{{ var("in_long_time") }}'::timestamp)
        as owner_to
        , **owner_id_changed**
    from event_first_owner
    where owner_id_changed is not null
```

### Expected Behaviour

L034 must be raised in the code provided in `What Happened` section.

### Observed Behaviour

After investigation it seems that it is due to the fact that we are using `'{{ var("in_long_time") }}'`. If we replace it by a traditional timestamp, the issue is raised. 
We are using the dbt templater. 

### How to reproduce

Execute the code with and without the Jinja variable

### Dialect

DBT templater

### Version

1.4.2

### Configuration

[sqlfluff]

dialect = redshift

exclude_rules = L026, L031, L032

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:rules]
max_line_length = 120

[sqlfluff:rules:L029]  # Keyword identifiers
ignore_words = name, year_month, state, type, stage, subscription, level, comment, event

[sqlfluff:rules:L010]
capitalisation_policy = lower

[sqlfluff:rules:L014]
extended_capitalisation_policy = lower

[sqlfluff:rules:L030]
extended_capitalisation_policy = lower

[sqlfluff:rules:L040]
capitalisation_policy = lower

[sqlfluff:rules:L063]
extended_capitalisation_policy = lower

[sqlfluff:rules:L066]
min_alias_length = 3

[sqlfluff:rules:L057]
quoted_identifiers_policy = none

[sqlfluff:rules:L059]
ignore_words = url, referrer, name

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pdebelak** [comment]
It seems like this is the case if `ignore_templated_areas` is set to `True` (the default). I _think_ the intention there is to not issue this warning where the whole select line is a jinja template, but I'm not 100% sure there. You can experiment with adding `ignore_templated_areas = False` to the `[sqlfluff]` section of your config in the meantime to see if that does what you would like.

**Comment by olivierdet** [comment]
Hello, thanks for the answer. It works but it cause a lot of new error message like 
`  47 | P:  10 | L059 | Unnecessary quoted identifier
                       | "stg_salesforce__emailmessage".`

**Comment by peterbud** [comment]
I don't see `stg_salesforce__emailmessage` in the above SQL statement. Can you provide a consistent repro pls?

**Comment by peterbud** [comment]
Closing due to lack of response
